### PR TITLE
Store: Add basic review listing screen

### LINKED ISSUE #17042: Store: Product Reviews: Add UI for listing and filtering reviews by status
Building off the state code added in #17041, a UI for viewing all of your product reviews is needed. 

This is the `Reviews List` screen from #12989. 

Functionally, this should work like our products & orders list tables including the search/filter bar and pagination.

### PR DESCRIPTION
**Pull request by justinshreve**
This PR implements a listing view based off of the design [here](https://automattic.invisionapp.com/share/R4BQDYYVY#/screens/251026936). It allows for filtering by status, has pagination, and allows you to search all reviews for a search term.

@kellychoffman @jameskoster This PR does have some CSS, however I expect some of the structure to change a bit as I start to implement the expanded/editor state of the reviews. Just a heads up. My main goal for this PR was to get the functionality working.

screen shot 2017-09-08 at 3 37 19 pm
screen shot 2017-09-08 at 3 38 14 pm
screen shot 2017-09-08 at 3 37 12 pm

Closes #17042.

To Test:
* Run `npm run test-client client/extensions/woocommerce/` and make sure all tests pass.
* You **must** use my `wc-api-dev` branch and not the production copy of `wc-api-dev`. Here is a copy: https://cloudup.com/cOW3IXvOy1u. Just download the zip, and upload it via the `wp-admin` plugin installer. You can do this on AT sites.
* Make sure you have a few reviews on your site. You can import the demo content, or post a few.
* Load up this branch.
* Go to `http://calypso.localhost:3000/store/reviews/:site` and verify that you see the filter/search bar.
* Select a tab that has reviews. Verify that the reviews show up correctly.
* Select a tab that has no reviews. Verify that you get an EmptyContent message.
* Enter a search term. You should see some results show up, or a EmptyContent message.
* (Optional) To test pagination if you don't have a ton of reviews, you can update  the `perPage` number in `reviews-list.js`, and the 'per_page' number in `/state/sites/reviews/utils.js`.

### CODE REVIEW

File: client/extensions/woocommerce/index.js [hunk 1]
```diff
@@ -112,6 +112,12 @@ const getStorePages = () => {
 			path: '/store/reviews/:site',
 		},
 		{
+			container: Reviews,
+			configKey: 'woocommerce/extension-reviews',
+			documentTitle: translate( 'Reviews' ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/index.js]: Hi! I've found a possible matching string that has already been translated **13** times:
`translate( 'Latest Reviews' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/reviews/reviews-list.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+			emptyMessage = translate( 'There are no reviews matching your search.' );
+			lineMessage = '';
+		} else if ( 'approved' === currentStatus ) {
+			emptyMessage = translate( 'No approved reviews.' );
+		} else if ( 'spam' === currentStatus ) {
+			emptyMessage = translate( 'No spam reviews.' );
+		} else if ( 'trash' === currentStatus ) {
+			emptyMessage = translate( 'No deleted reviews.' );
+		} else if ( 'any' === currentStatus ) {
+			emptyMessage = translate( 'No reviews.' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/reviews/reviews-list.js]: Hi! I've found a possible matching string that has already been translated **13** times:
`translate( 'Latest Reviews' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/reviews/reviews-list.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+					<div className="reviews__placeholder">
+					</div>
+				</Card>
+			);
+		} );
+	}
+
+	renderNoContent = () => {
+		const { currentSearch, currentStatus, translate } = this.props;
+		let emptyMessage = translate( 'No pending reviews.' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/reviews/reviews-list.js]: Hi! I've found a possible matching string that has already been translated **105** times:
`translate( 'Pending Review' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-review-list-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
One thought: I think we should get rid of the `All` view as its a bit redundant and Comments doesn't have it.

File: client/extensions/woocommerce/app/reviews/review-card.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	};
+
+	render() {
+		const { review } = this.props;
+		const classes = classNames( 'reviews__card', {
+			'is-approved': review.approved,
+			'is-unapproved': ! review.approved,
+		} );
+		const author = {
+			avatar_URL: review.avatar_urls[ 48 ],
```
Comment (allendav) [inline client/extensions/woocommerce/app/reviews/review-card.js]: Why this magic number? Maybe add a comment.

**Comment by allendav** [comment]
Idea: Show the number of pending reviews (if any) on the sidebar item.

**Comment by allendav** [comment]
When i switch back to a tab i’ve already been on, there is a delay - i would have expected the reviews to have already been loaded and display immediately. Maybe fix in a follow-on PR/issue.

File: client/extensions/woocommerce/app/reviews/reviews-filter-nav.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	clearSearch = () => {
+		const { status } = this.props;
+		this.props.updateCurrentReviewsQuery( this.props.site.ID, { search: '', status } );
+	}
+
+	render() {
+		const { translate, site, status } = this.props;
+		let currentSelection = translate( 'Pending' );
+
+		if ( 'approved' === status ) {
```
Comment (allendav) [inline client/extensions/woocommerce/app/reviews/reviews-filter-nav.js]: It is possible that someone might change the string to translate in the markup below but fail to update it here in this conditional. Please consider defining constants above for each of the translated strings and using them both here in this conditional and in the markup below.

Comment (justinshreve) [reply client/extensions/woocommerce/app/reviews/reviews-filter-nav.js]: This is the status returned by the API and is not translated. I could still define constants for these if that helps make it clearer.

**Comment by justinshreve** [comment]
> Maybe fix in a follow-on PR/issue.

Agreed, I can work on this in a separate PR to not hold up this stack of PRs. I'll open an issue.

**Comment by justinshreve** [comment]
> Idea: Show the number of pending reviews (if any) on the sidebar item.

#17044 😄

File: client/extensions/woocommerce/app/reviews/reviews-list.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+			status: currentStatus,
+		};
+		this.props.updateCurrentReviewsQuery( this.props.siteId, { page: 1, search: '' } );
+		if ( siteId ) {
+			this.props.fetchReviews( siteId, query );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const hasAnythingChanged = (
```
Comment (allendav) [inline client/extensions/woocommerce/app/reviews/reviews-list.js]: Optional. Destructuring props fields that are referenced multiple times would make this less verbose

**Pull request merged**